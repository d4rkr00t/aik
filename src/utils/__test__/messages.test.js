import stripAnsi from "strip-ansi";

import syntaxErrorMock from "./mock-data/syntax-error.json";
import buildStatsMock from "./mock-data/build-stats.json";

import {
  eslintExtraWarningMsg,
  fileDoesNotExistMsg,
  foundPackageJson,
  installingModuleMsg,
  packageJsonHasNotBeenFound,
  devServerBanner,
  devServerInvalidBuildMsg,
  devServerCompiledSuccessfullyMsg,
  devServerFailedToCompileMsg,
  devServerCompiledWithWarningsMsg,
  devServerFileDoesNotExistMsg,
  devServerRestartMsg,
  devServerModuleDoesntExists,
  devServerFrameworkDetectedRestartMsg,
  devServerReactRequired,
  builderBanner,
  builderRemovingDistMsg,
  builderRunningBuildMsg,
  builderErrorMsg,
  builderSuccessMsg,
  addTopSpace,
  addBottomSpace,
  joinWithSeparator
} from "../messages";

const print = msg => {
  // console.log(msg.join("\n"));
  return stripAnsi(msg.join("\n"));
};

const filename = "./src/index.js";

describe("Helpers", () => {
  test("#addTopSpace", () => {
    expect(addTopSpace(["msg", "content"]).join("")).toBe(["", "msg", "content"].join(""));
  });

  test("#addBottomSpace", () => {
    expect(addBottomSpace(["msg", "content"]).join("")).toBe(["msg", "content", ""].join(""));
  });

  describe("#joinWithSeparator", () => {
    test("array of string", () => {
      expect(joinWithSeparator("|", ["msg", "content", "here"]).join("")).toBe(
        ["msg", "|", "content", "|", "here"].join("")
      );
    });

    test("array of string arrays", () => {
      expect(joinWithSeparator("|", [["msg", "content"], ["here"]]).join("")).toBe(
        ["msg", "content", "|", "here"].join("")
      );
    });
  });
});

describe("Common Messages", () => {
  test("#eslintExtraWarningMsg", () => {
    expect(print(eslintExtraWarningMsg())).toMatchSnapshot();
  });

  test("#fileDoesNotExistMsg", () => {
    expect(print(fileDoesNotExistMsg(filename))).toMatchSnapshot();
  });

  test("#foundPackageJson", () => {
    expect(print(foundPackageJson())).toMatchSnapshot();
  });

  test("#installingModuleMsg", () => {
    expect(print(installingModuleMsg("react"))).toMatchSnapshot();
  });

  test("#packageJsonHasNotBeenFound", () => {
    expect(print(packageJsonHasNotBeenFound())).toMatchSnapshot();
  });
});

describe("Dev Server Messages", () => {
  describe("#devServerBanner", () => {
    let params;

    beforeEach(() => {
      params = {
        filename,
        host: "localhost",
        port: 3334,
        oldPort: 3333,
        framework: "react",
        template: {
          short: "index.html"
        },
        ngrok: "http://43kd92j3h.ngrok.com"
      };
    });

    test("all flags enabled", () => {
      expect(print(devServerBanner(params))).toMatchSnapshot();
    });

    test("no template", () => {
      params.template = {};
      expect(print(devServerBanner(params))).toMatchSnapshot();
    });

    test("no old port", () => {
      params.oldPort = false;
      expect(print(devServerBanner(params))).toMatchSnapshot();
    });

    test("no ngrok", () => {
      params.ngrok = false;
      expect(print(devServerBanner(params))).toMatchSnapshot();
    });
  });

  test("#devServerCompiledSuccessfullyMsg", () => {
    const params = {
      filename,
      host: "localhost",
      port: 3334,
      oldPort: 3333,
      framework: "react",
      template: {
        short: "index.html"
      },
      ngrok: "http://43kd92j3h.ngrok.com"
    };

    expect(print(devServerCompiledSuccessfullyMsg(params, 5800))).toMatchSnapshot();
  });

  test("#devServerCompiledWithWarningsMsg", () => {
    const params = {
      filename,
      host: "localhost",
      port: 3334,
      oldPort: 3333,
      framework: "react",
      template: {
        short: "index.html"
      },
      ngrok: "http://43kd92j3h.ngrok.com"
    };

    expect(print(devServerCompiledWithWarningsMsg(params, 5800))).toMatchSnapshot();
  });

  test("#devServerInvalidBuildMsg", () => {
    expect(print(devServerInvalidBuildMsg())).toMatchSnapshot();
  });

  test("#devServerFailedToCompileMsg", () => {
    expect(print(devServerFailedToCompileMsg())).toMatchSnapshot();
  });

  test("#devServerFileDoesNotExistMsg", () => {
    expect(print(devServerFileDoesNotExistMsg(filename))).toMatchSnapshot();
  });

  test("#devServerRestartMsg", () => {
    expect(print(devServerRestartMsg("react"))).toMatchSnapshot();
  });

  test("#devServerModuleDoesntExists", () => {
    expect(print(devServerModuleDoesntExists("react", filename))).toMatchSnapshot();
  });

  test("#devServerFrameworkDetectedRestartMsg", () => {
    expect(print(devServerFrameworkDetectedRestartMsg("react"))).toMatchSnapshot();
  });

  test("#devServerReactRequired", () => {
    expect(print(devServerReactRequired())).toMatchSnapshot();
  });
});

describe("Build Messages", () => {
  describe("#builderBanner", () => {
    let params;

    beforeEach(() => {
      params = {
        filename,
        base: "/subfolder/",
        template: {
          short: "index.html"
        }
      };
    });

    test("all flags and params", () => {
      expect(print(builderBanner(params))).toMatchSnapshot();
    });

    test("no template", () => {
      params.template = {};
      expect(print(builderBanner(params))).toMatchSnapshot();
    });

    test("no base", () => {
      params.base = "";
      expect(print(builderBanner(params))).toMatchSnapshot();
    });

    test("basic", () => {
      expect(print(builderBanner({ filename, template: {} }))).toMatchSnapshot();
    });
  });

  test("#builderRemovingDistMsg", () => {
    expect(print(builderRemovingDistMsg("./dist"))).toMatchSnapshot();
  });

  test("#builderRunningBuildMsg", () => {
    expect(print(builderRunningBuildMsg())).toMatchSnapshot();
  });

  describe("#builderErrorMsg?", () => {
    test("Custom Error message", () => {
      expect(print(builderErrorMsg(["Some error message"]))).toMatchSnapshot();
    });

    test("Syntax Error", () => {
      expect(print(builderErrorMsg(syntaxErrorMock))).toMatchSnapshot();
    });
  });

  test("#builderSuccessMsg", () => {
    expect(print(builderSuccessMsg("./dist", buildStatsMock))).toMatchSnapshot();
  });
});
