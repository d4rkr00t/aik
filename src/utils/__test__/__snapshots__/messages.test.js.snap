// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Build Messages #builderBanner all flags and params 1`] = `
" WAIT  Building...

Entry point:     ./src/index.js
Custom template: index.html
Base path:       /subfolder/"
`;

exports[`Build Messages #builderBanner basic 1`] = `
" WAIT  Building...

Entry point:     ./src/index.js"
`;

exports[`Build Messages #builderBanner no base 1`] = `
" WAIT  Building...

Entry point:     ./src/index.js
Custom template: index.html"
`;

exports[`Build Messages #builderBanner no template 1`] = `
" WAIT  Building...

Entry point:     ./src/index.js
Base path:       /subfolder/"
`;

exports[`Build Messages #builderErrorMsg? Custom Error message 1`] = `
" ERROR  Failed to create a production build. Reason:

Some error message"
`;

exports[`Build Messages #builderErrorMsg? Syntax Error 1`] = `
" ERROR  Failed to create a production build. Reason:

Error:

|  Syntax Error: Unexpected token (20:7)

Where: ./thinking-in-react/src/components/product-table/product-table.js

  18 |   return (
  19 |     <div className=\\"productTable\\">
> 20 |       <<table>
     |        ^
  21 |         <thead>
  22 |           <tr>
  23 |             <th>Name</th>"
`;

exports[`Build Messages #builderRemovingDistMsg 1`] = `"Removing folder: ./dist"`;

exports[`Build Messages #builderRunningBuildMsg 1`] = `"Running webpack production build..."`;

exports[`Build Messages #builderSuccessMsg 1`] = `
" DONE  in 3892ms

Aik has successfully generated a bundle in the \\"./dist\\" folder!
The bundle is optimized and ready to be deployed to production.

 ASSETS 

index.2b2f089a.js:  22.90kb, 7.33kb gzip
index.1027be30.css: 0.67kb, 0.31kb gzip
index.html:         0.34kb, 0.24kb gzip"
`;

exports[`Common Messages #eslintExtraWarningMsg 1`] = `
"You may use special comments to disable some warnings.
Use // eslint-disable-next-line to ignore the next line.
Use /* eslint-disable */ to ignore all warnings in a file."
`;

exports[`Common Messages #fileDoesNotExistMsg 1`] = `
" ERROR  File doesn't exist.

You are trying to use \\"./src/index.js\\" as an entry point, but this file doesn't exist.
Please, choose an existing file or create \\"./src/index.js\\" manualy."
`;

exports[`Common Messages #foundPackageJson 1`] = `
" WARNING  File \\"package.json\\" has been discovered.

Since \\"node_modules\\" folder doesn't exist and in order to avoid possible artifacts caused by
accidentally updated versions of npm modules Aik will run \\"npm install\\" in current directory.

 WAIT  Installing npm modules..."
`;

exports[`Common Messages #installingModuleMsg 1`] = `"Installing module \\"react\\" ..."`;

exports[`Common Messages #packageJsonHasNotBeenFound 1`] = `
" WARNING  File \\"package.json\\" hasn't been found.

In order to make subsequent builds more predictable Aik needs to create one.

 WAIT  Creating package.json..."
`;

exports[`Dev Server Messages #devServerBanner all flags enabled 1`] = `
"Entry point:      ./src/index.js
Custom template:  index.html
Port changed:      3333  ->  3334 
Server:           http://localhost:3334
Ngrok:            http://43kd92j3h.ngrok.com
React Hot Loader: enabled"
`;

exports[`Dev Server Messages #devServerBanner no ngrok 1`] = `
"Entry point:      ./src/index.js
Custom template:  index.html
Port changed:      3333  ->  3334 
Server:           http://localhost:3334
React Hot Loader: enabled"
`;

exports[`Dev Server Messages #devServerBanner no old port 1`] = `
"Entry point:      ./src/index.js
Custom template:  index.html
Server:           http://localhost:3334
Ngrok:            http://43kd92j3h.ngrok.com
React Hot Loader: enabled"
`;

exports[`Dev Server Messages #devServerBanner no react 1`] = `
"Entry point:      ./src/index.js
Custom template:  index.html
Port changed:      3333  ->  3334 
Server:           http://localhost:3334
Ngrok:            http://43kd92j3h.ngrok.com"
`;

exports[`Dev Server Messages #devServerBanner no template 1`] = `
"Entry point:      ./src/index.js
Port changed:      3333  ->  3334 
Server:           http://localhost:3334
Ngrok:            http://43kd92j3h.ngrok.com
React Hot Loader: enabled"
`;

exports[`Dev Server Messages #devServerCompiledSuccessfullyMsg 1`] = `
" DONE  Compiled successfully in 5800ms!

Entry point:      ./src/index.js
Custom template:  index.html
Port changed:      3333  ->  3334 
Server:           http://localhost:3334
Ngrok:            http://43kd92j3h.ngrok.com
React Hot Loader: enabled"
`;

exports[`Dev Server Messages #devServerCompiledWithWarningsMsg 1`] = `
" WARNING  Compiled with warnings in 5800ms.

Entry point:      ./src/index.js
Custom template:  index.html
Port changed:      3333  ->  3334 
Server:           http://localhost:3334
Ngrok:            http://43kd92j3h.ngrok.com
React Hot Loader: enabled

---------"
`;

exports[`Dev Server Messages #devServerFailedToCompileMsg 1`] = `" ERROR  Failed to compile."`;

exports[`Dev Server Messages #devServerFileDoesNotExistMsg 1`] = `" WARNING  File \\"./src/index.js\\" doesn't exist."`;

exports[`Dev Server Messages #devServerFrameworkDetectedRestartMsg 1`] = `
" WARNING  Usage of 'react' detected.

Restarting \\"webpack-dev-server\\"...

Please be patient and wait until restart completes, otherwise some changes might not be tracked."
`;

exports[`Dev Server Messages #devServerInvalidBuildMsg 1`] = `" WAIT  Compiling..."`;

exports[`Dev Server Messages #devServerModuleDoesntExists 1`] = `
" ERROR  Module 'react' doesn't exists.

Error in ./src/index.js

Webpack tried to resolve module  react  which doesn't exist.

It's likely caused by a typo in the module name."
`;

exports[`Dev Server Messages #devServerReactRequired 1`] = `
" WARNING  \\"react\\" required.

In order to make \\"react-hot-loader\\" work, \\"react\\" and \\"react-dom\\" are required.

 WAIT  Installing required modules..."
`;

exports[`Dev Server Messages #devServerRestartMsg 1`] = `
" WARNING  New npm module was added (react).

Restarting of the \\"webpack-dev-server\\" is requried.

Please be patient and wait until restart completes, otherwise some changes might not be tracked."
`;
